test/service/test_manifest.py

```diff
@@ -140,7 +142,23 @@ def _get_manifest_object(self, format_: ManifestFormat, filters: JSON) -> Manife
         service = ManifestService(StorageService())
         return service.get_manifest(format_=format_,
                                     catalog=self.catalog,
-                                    filters=filters)
+                                    filters=filters,
+                                    file_url_func=self.file_url)
+
+    def file_url(self,
+                 catalog: CatalogName,
+                 file_uuid: str,
+                 version: str,
+                 fetch: bool = True,
+                 **params) -> str:
+        file_uuid = urllib.parse.quote(file_uuid, safe='')
```

Comment:
do file_uuids contain unsafe characters?

does using furl to compose the full url w/ path provide escaping that you might need?
```
import uuid
import urllib
a = str(uuid.uuid4())
b = urllib.parse.quote(a)
b
'f15d144a-d86c-4fdd-8332-eced5d26b757'
a
'f15d144a-d86c-4fdd-8332-eced5d26b757'
import furl
furl.furl(url='https://azul.edu', path=a).url
'https://azul.edu/f15d144a-d86c-4fdd-8332-eced5d26b757'
furl.furl(url='https://azul.edu', path=f'{a}   hi').url
'https://azul.edu/f15d144a-d86c-4fdd-8332-eced5d26b757%20%20%20hi'
furl.furl(url='https://azul.edu', path=b).url
'https://azul.edu/f15d144a-d86c-4fdd-8332-eced5d26b757'
b+= '   hi'
furl.furl(url='https://azul.edu', path=b).url
'https://azul.edu/f15d144a-d86c-4fdd-8332-eced5d26b757%20%20%20hi'

Reply:
> file_uuids contain unsafe characters?

Not sure, but as you showed, `furl` handles the escaping we might need. When doing my changes I figure to leave it as close to the original as possible, so that I wouldn't break anything. 😅